### PR TITLE
docs(samples): create connection sample for MySQL instance

### PR DESCRIPTION
**Pull request by loferris**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/snippets/connection_create.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bigqueryconnection_connection_create]
+from google.cloud import bigquery_connection_v1 as bq_connection
+from google.cloud.bigquery_connection_v1 import types as connection_types
+
+"""This sample shows how to create a BigQuery connection with a Cloud MySql database"""
```
Comment (kurtisvg) [inline samples/snippets/connection_create.py:19]: nit: Should be "Cloud SQL for MySQL"

File: samples/snippets/connection_create.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    # [START_EXCLUDE]
+    original_project_id = project_id
+    original_location = location
+    original_connection_id = connection_id
+    original_database = database
+    original_instance = instance
+    original_instance_location = instance_location
+    original_username = username
+    original_password = password
+    # [END_EXCLUDE]
```
Comment (kurtisvg) [inline samples/snippets/connection_create.py:41]: What's up with these excludes and reassignments? Can we just leave this as is so the suer can edit it, and test the behavior of the function below?

File: samples/snippets/connection_create.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    cloud_sql_credential.username = username
+    cloud_sql_credential.password = password
+    cloud_sql_properties = bq_connection.CloudSqlProperties()
+    cloud_sql_properties.type_ = bq_connection.CloudSqlProperties.DatabaseType.MYSQL
+    cloud_sql_properties.database = database
+    cloud_sql_properties.instance_id = instance_id
+    cloud_sql_properties.credential = cloud_sql_credential
+    connection = connection_types.Connection()
+    connection.cloud_sql = cloud_sql_properties
+    create_connection(project_id, location, connection_id, connection)
```
Comment (kurtisvg) [inline samples/snippets/connection_create.py:84]: This is all stuff that should happen in the sample function below. 

nit: you may also want to consider using the mapping value here for these protos e.g.:
```python
cloud_sql_credential = bq_connection.CloudSqlCredential({
    "username": username,
    "password": password
})
```

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: samples/snippets/connection_create.py (hunk 2 above)
Comment (loferris) [reply samples/snippets/connection_create.py:41]: This is a pattern we've been following recently - I believe the purpose is to allow the function to be "live" and testable while still giving a clear view to the developer on how to fill in their own values. To me I think it makes more sense to have developer values set as statics above the function itself, but I'm not sure if that's consistent with what we've been doing. @tswast may have an opinion!

Comment (tswast) [reply samples/snippets/connection_create.py:41]: The `EXCLUDE` pattern is probably not necessary if the snippet includes the whole function definition, as you've done here. Using default values there would serve the same purpose.

Comment (tswast) [reply samples/snippets/connection_create.py:41]: Also, I tend to avoid `EXCLUDE` even with this pattern, as it shows "..." when rendered in the docs.

Comment (tswast) [reply samples/snippets/connection_create.py:41]: https://github.com/googleapis/python-datacatalog/blob/main/samples/snippets/data_catalog_ptm_create_taxonomy.py is a more typical example.

In core BigQuery we have a lot of samples that have the tags within the function, so we've kept with that style.

Comment (tswast) [reply samples/snippets/connection_create.py:41]: Slight update -- I've started to use the more standard style in some BigQuery samples: https://github.com/googleapis/python-bigquery/pull/1137

**Review by tswast — DISMISSED** [review]
+1 to Kurtis's requested changes.

File: samples/snippets/connection_create.py (hunk 2 above)
Comment (loferris) [reply samples/snippets/connection_create.py]: Attempted in recent commit! Would love feedback!

File: samples/snippets/connection_create.py (hunk 3 above)
Comment (loferris) [reply samples/snippets/connection_create.py]: resolved in recent commit!

File: samples/snippets/connection_create.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    return connection
+
+
+def create_connection(
```
[the remark this answers is not in the document]
Comment (loferris) [reply samples/snippets/connection_create.py]: resolved in recent commit!

File: samples/snippets/connection_create.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    cloud_sql_credential.password = password
+    cloud_sql_properties = bq_connection.CloudSqlProperties()
+    cloud_sql_properties.type_ = bq_connection.CloudSqlProperties.DatabaseType.MYSQL
+    cloud_sql_properties.database = database
+    cloud_sql_properties.instance_id = instance_id
+    cloud_sql_properties.credential = cloud_sql_credential
+    connection = connection_types.Connection()
+    connection.cloud_sql = cloud_sql_properties
+    create_connection(project_id, location, connection_id, connection)
+    return connection
```
[the remark this answers is not in the document]
Comment (loferris) [reply samples/snippets/connection_create.py]: resolved in recent commit!

File: samples/snippets/create_mysql_connection.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    cloud_sql_properties: bq_connection.CloudSqlProperties,
+) -> None:
+    connection = connection_types.Connection({"cloud_sql": cloud_sql_properties})
+    client = bq_connection.ConnectionServiceClient()
+    parent = client.common_location_path(project_id, location)
+    request = bq_connection.CreateConnectionRequest(
+        {"parent": parent, "connection": connection}
+    )
+    response = client.create_connection(request)
+    print(f"Created connection successfully: {response.name}")
```
Comment (kurtisvg) [inline samples/snippets/create_mysql_connection.py:94]: I'm finding this pattern incredibly confusing to try to navgate. My recommendation would be to do something like this:

```python
def main():
    // TODO(developer): Replace these variables before running the sample.
    project_id = "your-project-id"
    # See: https://cloud.google.com/bigquery/docs/locations for a list of available locations.
    location = "US"
    # The Cloud SQL instance connection name should be in the format "PROJECT:REGION:INSTANCE".
    cloud_sql_conn_name = "my-project:my-region:my-instance"
    # This is the name of the database on the Cloud SQL instance you want to connect to:
    db_name = "my-database"
    # This is the database user and password to connect to the database as:
    db_user = "my-username"
    db_pass = "my-password"
    create_mysql_connection(project_id, location, cloud_sql_conn_name, db_name, db_user, db_pass)

def create_mysql_connection(
    project_id: str,
    location: str,
    cloud_sql_conn_name: str,
    db_name: str,
    db_user: str, 
    db_pass = str,
  ):
    # Sample goes here

```

You can test directly on `create_mysql_connection` and still make clear to users what values need to be replaced.

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: samples/snippets/create_mysql_connection.py (hunk 6 above)
Comment (loferris) [reply samples/snippets/create_mysql_connection.py:94]: I prefer this pattern, for sure, will implement!

**Comment by loferris** [comment]
@kurtisvg All comments responded to!

File: samples/snippets/create_mysql_connection.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    database = "my-database"
+    # TODO(developer): Set instance to the instance where you're creating the connection.
+    instance_name = "my-instance"
+    # TODO(developer): Set instance_location to the location of the instance where you are creating the connection.
+    instance_location = "my-instance-location"
+    # TODO(developer): Set username to the database username.
+    username = "my-username"
+    # TODO(developer): Set password to the database password.
+    password = "my-password"
+    cloud_sql_conn_name = f"{project_id}:{instance_location}:{instance_name}"
```
Comment (kurtisvg) [inline samples/snippets/create_mysql_connection.py:39]: I don't think you need to have the user input these as 3 different fields. You can just ask them for the "instance connection name". This is a specific field in the instance metadata and is shown to the user in console, e.g.:
image


For reference, this is what we do in Cloud SQL samples:

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/HEAD/cloud-sql/mysql/sqlalchemy/main.py#L160-L173

**Review by kurtisvg — APPROVED** [review]

**Review by shollyman — APPROVED** [review]
I need to pay more attention to how we're dealing with test resource setup/lifecycle, but that is outside the scope of this PR.

**Pull request merged**